sprocs/assessment_instances_insert.sql

```diff
@@ -26,19 +28,61 @@ BEGIN
 
     -- ######################################################################
     -- determine the "number" of the new assessment instance
+    IF group_work THEN
+        SELECT gu.group_id
+        INTO  groupId
+        FROM  group_users as gu
+        JOIN  groups AS g ON(g.id = gu.group_id)
+        JOIN  group_configs AS gc ON (gc.id = g.group_config_id)
+        WHERE gu.user_id = assessment_instances_insert.user_id AND
+        gc.assessment_id = assessment_instances_insert.assessment_id AND
+        gc.deleted_at IS NULL AND
+        g.deleted_at IS NULL;
+        
+        IF assessment.multiple_instance THEN
+            SELECT coalesce(max(ai.number), 0) + 1
+            INTO number
+            FROM assessment_instances AS ai
+            WHERE
+                ai.assessment_id = assessment_instances_insert.assessment_id
+                AND ai.group_id = groupId;
+        END IF;
+        -- if a.multiple_instance is FALSE then we use
+        -- number = 1 so we will error on INSERT if there
+        -- are existing assessment_instances
+        
+        -- ######################################################################
+        -- do the actual insert
 
-    IF assessment.multiple_instance THEN
-        SELECT coalesce(max(ai.number), 0) + 1
-        INTO number
-        FROM assessment_instances AS ai
-        WHERE
-            ai.assessment_id = assessment_instances_insert.assessment_id
-            AND ai.user_id = assessment_instances_insert.user_id;
-    END IF;
 
-    -- if a.multiple_instance is FALSE then we use
-    -- number = 1 so we will error on INSERT if there
-    -- are existing assessment_instances
+        INSERT INTO assessment_instances
+                (auth_user_id, assessment_id, group_id, mode, auto_close, date_limit, number)
+        VALUES (authn_user_id, assessment_id, groupId, mode, auto_close, date_limit, number)
+        RETURNING id
+        INTO assessment_instance_id;
+    ELSE 
+         IF assessment.multiple_instance THEN
+            SELECT coalesce(max(ai.number), 0) + 1
+            INTO number
+            FROM assessment_instances AS ai
+            WHERE
+                ai.assessment_id = assessment_instances_insert.assessment_id
+                AND ai.user_id = assessment_instances_insert.user_id;
+        END IF;
+        -- if a.multiple_instance is FALSE then we use
+        -- number = 1 so we will error on INSERT if there
+        -- are existing assessment_instances
+        
+        -- ######################################################################
+        -- do the actual insert
+
+
```

Comment:
The formatting convention seems to only use one blank line after comments like this?